code/modules/food/food/snacks.dm

```diff
@@ -384,43 +411,256 @@
+
+/obj/item/weapon/reagent_containers/food/snacks/donut/yellow/jelly
+	name = "yellow frosted jelly donut"
+	icon_state = "jelly_yellow"
+	desc = "This one has yellow frosting and a jelly filling!"
+/obj/item/weapon/reagent_containers/food/snacks/donut/yellow/jelly/Initialize()
+	. = ..()
+	reagents.add_reagent("nutriment", 3, nutriment_desc)
+	reagents.add_reagent("berryjuice", 5)
+
+/obj/item/weapon/reagent_containers/food/snacks/donut/olive
+	name = "olive frosted donut"
+	icon_state = "donut_olive"
+	desc = "This one has olive frosting!"
+	overlay_state = "donut_olive_inbox"
+/obj/item/weapon/reagent_containers/food/snacks/donut/olive/Initialize()
+	. = ..()
+	reagents.add_reagent("nutriment", 3, nutriment_desc)
+
+/obj/item/weapon/reagent_containers/food/snacks/donut/olive/jelly
+	name = "olive frosted jelly donut"
+	icon_state = "jelly_olive"
+	desc = "This one has olive frosting and a jelly filling!"
+/obj/item/weapon/reagent_containers/food/snacks/donut/olive/jelly/Initialize()
+	. = ..()
+	reagents.add_reagent("nutriment", 3, nutriment_desc)
+	reagents.add_reagent("berryjuice", 5)
+
+/obj/item/weapon/reagent_containers/food/snacks/donut/homer
+	name = "frosted donut with sprinkles"
+	icon_state = "donut_homer"
+	desc = "It's a d'ohnut!"
+	overlay_state = "donut_homer_inbox"
+/obj/item/weapon/reagent_containers/food/snacks/donut/homer/Initialize()
+	. = ..()
+	reagents.add_reagent("nutriment", 3, nutriment_desc)
+	reagents.add_reagent("sprinkles", 1)
+
+/obj/item/weapon/reagent_containers/food/snacks/donut/homer/jelly
+	name = "frosted jelly donut with sprinkles"
+	icon_state = "jelly_homer"
+	desc = "It's a d'ohnut with jelly filling!"
+/obj/item/weapon/reagent_containers/food/snacks/donut/homer/jelly/Initialize()
+	. = ..()
+	reagents.add_reagent("nutriment", 3, nutriment_desc)
+	reagents.add_reagent("sprinkles", 1)
+	reagents.add_reagent("berryjuice", 5)
+
+/obj/item/weapon/reagent_containers/food/snacks/donut/choc_sprinkles
+	name = "chocolate sprinkles donut"
+	icon_state = "donut_choc_sprinkles"
+	desc = "Mmm, chocolate with sprinkles... approaching maximum donut."
+	overlay_state = "donut_choc_sprinkles_inbox"
+/obj/item/weapon/reagent_containers/food/snacks/donut/choc_sprinkles/Initialize()
+	. = ..()
+	reagents.add_reagent("nutriment", 3, nutriment_desc)
+	reagents.add_reagent("sprinkles", 1)
+
+/obj/item/weapon/reagent_containers/food/snacks/donut/choc_sprinkles/jelly
+	name = "chocolate sprinkles jelly donut"
+	icon_state = "jelly_choc_sprinkles"
+	desc = "Pretty sure this is the most sugar you can pack into a donut."
+/obj/item/weapon/reagent_containers/food/snacks/donut/choc_sprinkles/jelly/Initialize()
 	. = ..()
 	reagents.add_reagent("nutriment", 3, nutriment_desc)
 	reagents.add_reagent("sprinkles", 1)
-	if(prob(30))
-		src.icon_state = "donut2"
-		src.overlay_state = "box-donut2"
-		src.name = "frosted donut"
-		reagents.add_reagent("sprinkles", 2)
-		center_of_mass = list("x"=19, "y"=16)
```

Comment:
Why was this chance at a frosted donut removed? This isn't documented and essentially reduces frosted donuts to the realm of security donut box RNG and donut vending machines. Chefs should be able to make these, and this implementation is kinda fun. Sprinkles are also only obtainable this way, and required for certain recipes.

Reply:
You should be able to get sprinkles another way, and specifically make donuts with sprinkles. Rng shapes for chicken nuggets is aggravating enough as it is.

Reply:
As said in PR, part of the point is that A) a new recipe for sprinkles should be made down the line, not be a random RNG donut variety and B) all new donuts should have proper specific recipies of their own, not be entirely RNG. It literally says so in description.